HostLibraryTests/DecisionTree_test.cpp

```diff
@@ -369,6 +369,8 @@ TEST(DecisionTree, DecisionTreeBatch)
     using SizeInRange = Predicates::Contraction::SizeInRange;
     using Range       = Predicates::Contraction::Range;
     using And         = Predicates::And<ContractionProblem>;
+    using Key         = std::array<float, 4>;
+    using DTree       = Tree<Key, std::shared_ptr<ContractionLibrary>, std::shared_ptr<ContractionSolution>>;    
```

Comment:
I see that previously we were pushing an array of size 4 (L413) into a container that's defined at file-scope to have size 3 (L40). Odd that other platforms aren't catching this error. However, I also notice this behaviour at L294, L313, etc. Should lines these be updated as well? Why aren't they failing too? Which line did the original error come from?

Reply:
The important distinction between this change and the others is that in the previous test the size of the features vector is three (note that we call `push_back` three times). In this test the size of features is four. That is why it was failing prior to this change. Why it wasn't failing on other platforms (shrug... undefined behavior).

Reply:
Thanks for your comment.
Maybe better to put some comment in source code.
Do we need to change other locations using Key?

Reply:
A comment in the test code would probably help a little. We do not need to change the other `Key`s but I think I will move the aliases into the tests where they are used to avoid confusion.